src/io/paf.rs

```diff
@@ -0,0 +1,323 @@
+    }
+}
+
+
+/// A PAF record.
+pub struct Records<'a, R: 'a + io::Read> {
+    inner: csv::DeserializeRecordsIter<'a, R, (String, u64, u64, u64, char, String, u64, u64, u64, u64, u64, u8)>,
+}
+
+
+impl<'a, R: io::Read> Iterator for Records<'a, R> {
+    type Item = csv::Result<Record>;
+
+    fn next(&mut self) -> Option<csv::Result<Record>> {
+        self.inner
+            .next()
+            .map(|res| {
+                res.map(|(name_query, length_query, begin_query, end_query, strand, name_target, length_target, begin_target, end_target, base_match, mapping_length, score)| {
+                            Record {
+                            name_query:       name_query,
+                            length_query:     length_query,
+                            begin_query:      begin_query,
+                            end_query:        end_query,
+                            strand:         strand,
+                            name_target:    name_target,
+                            length_target:  length_target,
+                            begin_target:   begin_target,
+                            end_target:     end_target,
+                            base_match:     base_match,
+                            mapping_length: mapping_length,
+                            score:          score,
+                            }
+                        })
+            })
+    }
+}
+
+
+/// A PAF writer.
+pub struct Writer<W: io::Write> {
+    inner: csv::Writer<W>,
+}
+
+
+impl Writer<fs::File> {
+    /// Write to a given file path.
+    pub fn to_file<P: AsRef<Path>>(path: P) -> io::Result<Self> {
+        fs::File::create(path).map(Writer::new)
+    }
+}
+
+
+impl<W: io::Write> Writer<W> {
+    /// Write to a given writer.
+    pub fn new(writer: W) -> Self {
+        Writer {
+            inner: csv::WriterBuilder::new()
+                .has_headers(false)
+                .delimiter(b'\t')
+                .flexible(true)
+                .from_writer(writer),
+        }
+    }
+
+    /// Write a given PAF record.
+    pub fn write(&mut self, record: &Record) -> csv::Result<()> {
+        self.inner.serialize(record)
+    }
+}
+
+
+/// PAF record as defined in "Minimap and miniasm: fast mapping and de novo assembly for noisy long sequences" Heng Li 2017
+/// https://academic.oup.com/bioinformatics/article/32/14/2103/1742895
+/// Specification : https://lh3.github.io/minimap2/minimap2.html
+#[derive(Debug, Serialize, Default)]
+pub struct Record {
+    name_query:       String,
+    length_query:     u64,
+    begin_query:      u64,
+    end_query:        u64,
+    strand:    char,
+    name_target:    String,
+    length_target:  u64,
+    begin_target:   u64,
+    end_target:     u64,
+    base_match:     u64, 
+    mapping_length: u64,
+    score:          u8,
+}
+
+impl Record {
+    /// Create a new PAF record.
+    pub fn new() -> Self {
+        Record {
+            name_query:       "".to_owned(),
```

Comment:
I think it would make more sense to have a separate struct 
```
Locus {
    name: ...
    length: ...
    begin: ...
    end: ...
```

And then use that within record, i.e.
```
Record {
    query: Locus,
    target: Locus,
    strand: char,
    score: u8,
    ...
}
```